pkg/git/repository.go

```diff
@@ -244,6 +244,13 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
 		}
 	}
 
+	if o.CloneForWrite {
```

Comment:
If we created the repo in line 230 there is no point in validating write permission. Please take it into consideration.